pennylane/interfaces/jacobian_products.py

```diff
@@ -0,0 +1,176 @@
+        >>> vjps = jp_method.compute_vjp(batch, dys)
+        >>> vjps
+        (array([-0.04991671]), array([2.54286107]))
+        >>> expected_vjp0 = 0.5 * -np.sin(0.1)
+        >>> qml.math.allclose(vjps[0], expected_vjp0)
+        True
+        >>> expected_jvp1 = 2.0 * -np.sin(0.2) + 3.0 * np.cos(0.2)
+        >>> qml.math.allclose(vjps[1], expected_vjp1)
+        True
+
+        """
+
+    @abc.abstractmethod
+    def compute_jacobian(self, tapes: Batch) -> Tuple:
+        """Compute the full jacobian for a batch of tapes.
+
+        This method is required to compute Jacobians in the ``jax-jit`` interface
+
+        Args:
+            tapes: the batch of tapes to take the jacobian of
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> jp_method.compute_jacobian(batch)
+        (array(-0.09983342), (array(-0.19866933), array(0.98006658)))
+
+        """
+
+
+class TransformJacobianProducts(JacobianProductCalculator):
+    """Compute VJPs, JVPs and Jacobians via a :class:`~.gradient_transform`.
+
+    Args:
+        inner_execute (Callable[[Tuple[QuantumTape]], ResultBatch]): a function that
+            executes the batch of circuits and returns their results.
+        gradient_transform (pennylane.gradients.gradient_transform): the gradient transform to use.
+        gradient_kwargs (dict): Any keyword arguments for the gradient transform.
+
+    >>> inner_execute = qml.device('default.qubit').execute
+    >>> gradient_transform = qml.gradients.param_shift
+    >>> kwargs = {"broadcast": True}
+    >>> jp_method = TransformJacobianProducts(inner_execute, gradient_transform, kwargs)
+
+    """
+
+    def __repr__(self):
+        return f"TransformJacobianProducts({self._inner_execute}, gradient_transform={self._gradient_transform}, gradient_kwargs={self._gradient_kwargs})"
+
+    def __init__(
+        self,
+        inner_execute: Callable,
+        gradient_transform: "qml.gradients.gradient_transform",
+        gradient_kwargs: Optional[dict] = None,
+    ):
+        self._inner_execute = inner_execute
+        self._gradient_transform = gradient_transform
+        self._gradient_kwargs = gradient_kwargs or {}
+
+    def execute_and_compute_jvp(self, tapes: Batch, tangents: Tuple[Tuple[TensorLike]]):
+        num_result_tapes = len(tapes)
+
+        jvp_tapes, jvp_processing_fn = qml.gradients.batch_jvp(
+            tapes, tangents, self._gradient_transform, gradient_kwargs=self._gradient_kwargs
+        )
+
+        full_batch = tapes + tuple(jvp_tapes)
+
+        full_results = self._inner_execute(full_batch)
+
+        results = full_results[:num_result_tapes]
+        jvp_results = full_results[num_result_tapes:]
+        jvps = jvp_processing_fn(jvp_results)
+        return tuple(results), tuple(jvps)
```

Comment:
Not sure how important this is, but `jvp_processing_fn` already casts the `jvps` to `tuple`.

Reply:
```
>>> a = (1,)
>>> b = tuple(a)
>>> a is b
True
```

So we don't need to worry about making a copy if its already a tuple. A lot of things return sometimes lists, sometimes tuples, so I like to ensure everything is tuples when it is in my power to do so.

Reply:
Ah, I understand. Yeah, that sounds like a good policy! Thanks! :)